### ISSUE TITLE
General_playback_error

### ISSUE THREAD

**Issue by chrisboer** [matjaz/homey-youtube#12]
![b7ebd414-0f3f-11e7-920b-25c4faa49b0e](https://cloud.githubusercontent.com/assets/23163409/24293310/6bca3d0c-1091-11e7-8ecc-55242b92f8e1.png)

https://github.com/athombv/homey/issues/1393

**Comment by TheMSB** [comment]
@matjaz If it helps here is a track that will reproduce the error without fail:

Search for 'naturally'
```
{ track_id: 'rMcKvp_Glb8',
  title: 'Animal Adventure Park Giraffe Cam',
  artist: [ { name: 'Animal Adventure Park', type: 'artist' } ],
  album: false,
  duration: false,
  artwork: 
   { small: 'https://i.ytimg.com/vi/rMcKvp_Glb8/default_live.jpg',
     medium: 'https://i.ytimg.com/vi/rMcKvp_Glb8/mqdefault_live.jpg',
     large: 'https://i.ytimg.com/vi/rMcKvp_Glb8/hqdefault_live.jpg' },
  genre: false,
  release_date: false,
  bitrate: false,
  codecs: [ 'homey:codec:mp3' ],
  bpm: false,
  options: false,
  player_uri: 'homey:app:com.youtube',
  app_icon: '/app/com.youtube/assets/icon.svg',
  playCount: 0,
  skipCount: 0,
  lastPlayed: false,
  lastSkipped: false,
  regCount: 0 }
```
The Youtube app returns an error when Homey asks for a stream_url:
` { [Error: Unexpected token R] message: 'Unexpected token R' } undefined`

**Comment by matjaz** [comment]
@MarvinSchenkel can you please take a look?

**Comment by chrisboer** [comment]
I get this error 
{
 "error": {
  "errors": [
   {
    "domain": "usageLimits",
    "reason": "dailyLimitExceededUnreg",
    "message": "Daily Limit for Unauthenticated Use Exceeded. Continued use requires signup.",
    "extendedHelp": "https://code.google.com/apis/console"
   }
  ],
  "code": 403,
  "message": "Daily Limit for Unauthenticated Use Exceeded. Continued use requires signup."
 }
}

**Comment by MarvinSchenkel** [comment]
Looks like you have to setup your own api key. Go to settings to register one. I am looking into implementing logging in to YouTube so we won't encounter this error anymore

**Comment by chrisboer** [comment]
@MarvinSchenkel I have a own api key so with this link it works with https://www.youtube.com/watch?v=f4Mc-NYPHaQ&ytbChannel=Queen%20Official

but it does not work with the first one if you type in queen in the search of "Muziek" and try to play it you get generic_playback_issue which is this one 
https://www.youtube.com/watch?v=_Uu12zY01ts&ytbChannel=Queen%20Official

**Comment by MarvinSchenkel** [comment]
@chrisboer Hm, starting to look like it has to do with long videos (> 1h). Can you confirm you get the error with relative long videos?

**Comment by MarvinSchenkel** [comment]
Yeah figured it out. It has to do with the youtube>mp3 library we use to create the stream url. I get the following error if I manually try to convery the video on their website:

_There was an Error caused by YouTube, we cannot deliver this Video!
This error is mostly caused by copyright issues or the length of the video.
We only support videos with maximum of 20 minutes._

@matjaz Any idea for a better youtube>mp3 converter?

**Comment by chrisboer** [comment]
Yes with long videos but also with commercial video's (short) like if you want to play "naturally" from the artist Ronde you also get a generic_playback_error
https://www.youtube.com/watch?v=WMaq2-w5EAw

**Comment by matjaz** [comment]
This one looks good http://www.youtubeinmp3.com/api/

**Comment by MarvinSchenkel** [comment]
@matjaz Implemented this one. Looks like this one has a 120 minutes limit, should be alright for most videos. I have look for other ones, but could not find one that converts videos that are longer than 120 minutes. Looks like Homey does not pick up any errors thrown by the app, but it always displays 'generic_playback_error'.

**Comment by TheMSB** [comment]
@MarvinSchenkel A user friendly approach would be to filter any tracks that exceed this limit when responding to media search queries.

**Comment by MarvinSchenkel** [comment]
@TheMSB Thanks, really good idea. Fixed it and implemented the filter. The duration is now also shown in the search results :). Will be committed as part of the next release. I am still working on the login so I can fetch personal playlists.

**Comment by matjaz** [comment]
filter is necessary only for local mp3 playback (requiring converting service). chromecast and others which stream video directly can handle longer videos.

**Comment by TheMSB** [comment]
> filter is necessary only for local mp3 playback (requiring converting service). chromecast and others which stream video directly can handle longer videos.

Exactly, it would make sense to implement this filter only when responding to search/play requests made by Homey media.

**Comment by MarvinSchenkel** [comment]
Of course I only applied the filter for homey media playback ;)

**Comment by matjaz** [comment]
version ff345b71b682add43061b7948a2b488be3f6eb06 should fix this
